File: web/src/scenarios/manifest.ts
/**
 * Scenario registry (UI) — mirrors shared/scenarios/manifest.py.
 * Used by dashboard dropdown labels, topology strip, landing System Map.
 */

export type ScenarioTier = 'hero' | 'research' | 'sponsor' | 'ops'

export interface AgentNode {
  id: string
  label: string
  langgraphNode: string
  role: string
}

export interface TopologyEdge {
  from: string
  to: string
  payload: string
}

export interface ScenarioManifest {
  id: string
  label: string
  tier: ScenarioTier
  tagline: string
  task: string
  agents: AgentNode[]
  edges: TopologyEdge[]
  parallel: string[][]
  /** Primary fault — root replay must flip oracle */
  fault: { agent: string; field: string; symptom: string }
  /** Decoy — replay must NOT flip (demo beat) */
  decoy: { agent: string; field: string; symptom: string }
  arize: { fail: string; healed: string; evaluators: string[] }
}

export const SCENARIO_MANIFEST: ScenarioManifest[] = [
  {
    id: 'claim_adjudication',
    label: 'insurance · claim adjudication',
    tier: 'hero',
    tagline: 'INTAKE decimal slip → $52k payout — fix ADJUSTER first does not flip.',
    task: 'Adjudicate member claim · MRI lumbar · gold tier',
    agents: [
      { id: 'intake', label: 'INTAKE', langgraphNode: 'intake', role: 'Parse claim → structured hand-off' },
      { id: 'coverage', label: 'COVERAGE', langgraphNode: 'coverage', role: 'Tier limits + procedure coverage' },
      { id: 'fraud', label: 'FRAUD', langgraphNode: 'fraud', role: 'Parallel anomaly scan' },
      { id: 'adjuster', label: 'ADJUSTER', langgraphNode: 'adjuster', role: 'Merge → payout decision' },
      { id: 'payout', label: 'PAYOUT', langgraphNode: 'payout', role: 'Execute payment' },
    ],
    edges: [
      { from: 'intake', to: 'coverage', payload: '{member_id, procedure, billed_amount, policy_tier}' },
      { from: 'intake', to: 'fraud', payload: '{member_id, procedure, billed_amount, policy_tier}' },
      { from: 'coverage', to: 'adjuster', payload: '{approved_amount, tier_limit}' },
      { from: 'fraud', to: 'adjuster', payload: '{risk_score, flags[]}' },
      { from: 'adjuster', to: 'payout', payload: '{payout_amount}' },
    ],
    parallel: [['coverage', 'fraud']],
    fault: {
      agent: 'intake',
      field: 'billed_amount',
      symptom: 'Payout $52,000 exceeds gold tier limit',
    },
    decoy: {
      agent: 'adjuster',
      field: 'payout_amount',
      symptom: 'Adjuster trim plausible — replay does not flip',
    },
    arize: {
      fail: 'claim_fail',
      healed: 'claim_healed',
      evaluators: ['payout_within_policy', 'root_step_is_intake_not_adjuster', 'replay_flipped'],
    },
  },
  {
    id: 'prior_auth',
    label: 'clinical · prior authorization',
    tier: 'research',
    tagline: 'Chart reader wrong date → guideline pass → submission denied.',
    task: 'Prior auth lumbar MRI · UHC · chart CH-9921',
    agents: [
      { id: 'chart_reader', label: 'CHART_READER', langgraphNode: 'chart_reader', role: 'Clinical fact extraction' },
      { id: 'guideline_check', label: 'GUIDELINE_CHECK', langgraphNode: 'guideline_check', role: 'Medical necessity rules' },
      { id: 'auth_writer', label: 'AUTH_WRITER', langgraphNode: 'auth_writer', role: 'PA letter + citations' },
      { id: 'submit', label: 'SUBMIT', langgraphNode: 'submit', role: 'Payer portal submission' },
    ],
    edges: [
      { from: 'chart_reader', to: 'guideline_check', payload: '{diagnosis, prior_treatments, contraindications}' },
      { from: 'guideline_check', to: 'auth_writer', payload: '{meets_criteria, rule_ids[]}' },
      { from: 'auth_writer', to: 'submit', payload: '{auth_payload}' },
    ],
    parallel: [],
    fault: {
      agent: 'chart_reader',
      field: 'prior_treatments',
      symptom: 'Denied — insufficient therapy interval (90-day rule)',
    },
    decoy: {
      agent: 'auth_writer',
      field: 'auth_payload',
      symptom: 'Letter wording — replay does not flip denial',
    },
    arize: {
      fail: 'pa_fail',
      healed: 'pa_healed',
      evaluators: ['guideline_grounded_in_chart', 'earliest_fault_before_symptom', 'replay_flipped'],
    },
  },
  {
    id: 'procurement_gpu',
    label: 'procurement · GPU purchase',
    tier: 'sponsor',
    tagline: 'Browser mis-read SKU — cheapest quote is wrong variant.',
    task: '500× H100-80GB under $25k · Browserbase vendor scrape',
    agents: [
      { id: 'spec', label: 'SPEC', langgraphNode: 'spec', role: 'Purchase constraints' },
      { id: 'browser', label: 'BROWSER', langgraphNode: 'browser', role: 'Live page scrape (Browserbase)' },
      { id: 'compare', label: 'COMPARE', langgraphNode: 'compare', role: 'Rank quotes vs spec' },
      { id: 'approver', label: 'APPROVER', langgraphNode: 'approver', role: 'Delegation sign-off' },
      { id: 'po', label: 'PO', langgraphNode: 'po', role: 'Issue purchase order' },
    ],
    edges: [
      { from: 'spec', to: 'browser', payload: '{sku, qty, max_price, vendor_allowlist}' },
      { from: 'browser', to: 'compare', payload: '{sku_selected, unit_price, page_snapshot}' },
      { from: 'compare', to: 'approver', payload: '{best_offer}' },
      { from: 'approver', to: 'po', payload: '{approved, best_offer}' },
    ],
    parallel: [],
    fault: {
      agent: 'browser',
      field: 'sku_selected',
      symptom: 'Oracle FAIL: SKU mismatch (refurb ≠ new)',
    },
    decoy: {
      agent: 'compare',
      field: 'best_offer',
      symptom: 'Gray-market vendor — fix compare alone no flip',
    },
    arize: {
      fail: 'procurement_fail',
      healed: 'procurement_healed',
      evaluators: ['selected_sku_matches_spec', 'replay_flipped'],
    },
  },
  {
    id: 'soc_incident',
    label: 'security · SOC incident',
    tier: 'ops',
    tagline: 'Wrong IOC → contain decoy host — threat still active.',
    task: 'SIEM alert #8812 · suspected C2 beacon',
    agents: [
      { id: 'alert_triage', label: 'ALERT_TRIAGE', langgraphNode: 'alert_triage', role: 'Classify alert + IOC' },
      { id: 'threat_intel', label: 'THREAT_INTEL', langgraphNode: 'threat_intel', role: 'Parallel feed enrichment' },
      { id: 'enrich', label: 'ENRICH', langgraphNode: 'enrich', role: 'Merge → target hosts' },
      { id: 'contain', label: 'CONTAIN', langgraphNode: 'contain', role: 'Isolation playbooks' },
      { id: 'notify', label: 'NOTIFY', langgraphNode: 'notify', role: 'Page + ticket' },
    ],
    edges: [
      { from: 'alert_triage', to: 'enrich', payload: '{severity, ioc, affected_hosts}' },
      { from: 'alert_triage', to: 'threat_intel', payload: '{ioc}' },
      { from: 'threat_intel', to: 'enrich', payload: '{ioc_verdict}' },
      { from: 'enrich', to: 'contain', payload: '{confirmed_hosts}' },
      { from: 'contain', to: 'notify', payload: '{isolated, runbook_id}' },
    ],
    parallel: [['threat_intel']],
    fault: {
      agent: 'alert_triage',
      field: 'ioc',
      symptom: 'Contained wrong host — C2 still beaconing',
    },
    decoy: {
      agent: 'contain',
      field: 'isolated',
      symptom: 'Playbook ran — fix contain alone no flip',
    },
    arize: {
      fail: 'soc_fail',
      healed: 'soc_healed',
      evaluators: ['containment_matches_true_ioc', 'replay_flipped'],
    },
  },
]

export const HERO_SCENARIO_ID = 'claim_adjudication'
export const RESEARCH_SCENARIO_ID = 'prior_auth'

export function manifestById(id: string): ScenarioManifest | undefined {
  return SCENARIO_MANIFEST.find((s) => s.id === id)
}

File: shared/scenarios/manifest.py
"""Canonical scenario registry — four demo domains replacing legacy AP/claims.

Each domain defines:
  - agent topology (LangGraph nodes + parallel branches)
  - structured hand-off payloads (what the dashboard topology strip renders)
  - injected fault + decoy (for the replay climax: decoy no-flip → root flip)
  - oracle + Arize evaluators (sponsor / research track)

UI labels and graph layout: web/src/scenarios/manifest.ts (kept in sync).
"""
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any, Literal

Tier = Literal["hero", "research", "sponsor", "ops"]


@dataclass(frozen=True)
class HandoffField:
    name: str
    dtype: str
    example: str
    description: str = ""


@dataclass(frozen=True)
class AgentSpec:
    id: str
    label: str
    langgraph_node: str
    role: str
    emits: tuple[HandoffField, ...]


@dataclass(frozen=True)
class EdgeSpec:
    src: str
    dst: str
    payload: str  # human-readable hand-off name shown in topology


@dataclass(frozen=True)
class FaultSpec:
    agent: str
    field: str
    bad_value: Any
    good_value: Any
    symptom: str  # what the oracle / final agent surfaces


@dataclass(frozen=True)
class LangGraphSpec:
    checkpointer: str
    fork_api: str
    replay_api: str
    state_channels: tuple[str, ...]


@dataclass(frozen=True)
class ArizeSpec:
    fail_trace_id: str
    healed_trace_id: str
    evaluators: tuple[str, ...]
    span_tags: tuple[str, ...]
    improvement_loop: str


@dataclass(frozen=True)
class ScenarioDomain:
    id: str
    label: str
    tier: Tier
    tagline: str
    task_template: str
    agents: tuple[AgentSpec, ...]
    edges: tuple[EdgeSpec, ...]
    parallel_groups: tuple[tuple[str, ...], ...]
    primary_fault: FaultSpec
    decoy_fault: FaultSpec
    oracle: str
    langgraph: LangGraphSpec
    arize: ArizeSpec
    research_notes: str = ""


# ---------------------------------------------------------------------------
# 1. Insurance claim adjudication (HERO — landing + booth default)
# ---------------------------------------------------------------------------
_INSURANCE_AGENTS = (
    AgentSpec(
        "intake", "INTAKE", "intake", "Parse member claim form → structured hand-off",
        (
            HandoffField("member_id", "str", "M-8842", "Unique member identifier"),
            HandoffField("procedure", "str", "MRI lumbar", "CPT / procedure code"),
            HandoffField("billed_amount", "float", "5200.00", "Provider billed amount (USD)"),
            HandoffField("policy_tier", "enum", "gold", "gold | silver | platinum"),
        ),
    ),
    AgentSpec(
        "coverage", "COVERAGE", "coverage", "Verify procedure covered under tier limits",
        (HandoffField("approved_amount", "float", "5200.00", "Tier-adjusted approved cap"),),
    ),
    AgentSpec(
        "fraud", "FRAUD", "fraud", "Parallel anomaly scan (can flag but be overridden)",
        (HandoffField("risk_score", "float", "0.12", "0–1; high triggers review"),),
    ),
    AgentSpec(
        "adjuster", "ADJUSTER", "adjuster", "Merge coverage + fraud → payout decision",
        (HandoffField("payout_amount", "float", "5200.00", "Amount sent to payment rail"),),
    ),
    AgentSpec(
        "payout", "PAYOUT", "payout", "Execute payment + notify member",
        (HandoffField("paid", "bool", "True", "Oracle reads this + policy cap"),),
    ),
)

INSURANCE = ScenarioDomain(
    id="claim_adjudication",
    label="insurance · claim adjudication",
    tier="hero",
    tagline="We almost paid $52k instead of $5.2k — root cause was INTAKE, not ADJUSTER.",
    task_template="Adjudicate member claim M-8842 · MRI lumbar · billed ${amount}",
    agents=_INSURANCE_AGENTS,
    edges=(
        EdgeSpec("intake", "coverage", "{member_id, procedure, billed_amount, policy_tier}"),
        EdgeSpec("intake", "fraud", "{member_id, procedure, billed_amount, policy_tier}"),
        EdgeSpec("coverage", "adjuster", "{approved_amount, tier_limit}"),
        EdgeSpec("fraud", "adjuster", "{risk_score, flags[]}"),
        EdgeSpec("adjuster", "payout", "{payout_amount, member_id}"),
    ),
    parallel_groups=(("coverage", "fraud"),),
    primary_fault=FaultSpec(
        agent="intake",
        field="billed_amount",
        bad_value=52000.00,
        good_value=5200.00,
        symptom="Payout $52,000 exceeds gold tier policy limit ($8,000)",
    ),
    decoy_fault=FaultSpec(
        agent="adjuster",
        field="payout_amount",
        bad_value=4800.00,
        good_value=5200.00,
        symptom="Adjuster trimmed payout — plausible but not root; replay does not flip",
    ),
    oracle="payout_within_policy(member_id, procedure, amount, tier)",
    langgraph=LangGraphSpec(
        checkpointer="MemorySaver",
        fork_api='update_state(as_node="intake", values={"billed_amount": 5200.0})',
        replay_api="graph.invoke(None, config) → re-run from checkpoint",
        state_channels=("claim", "coverage", "fraud", "decision", "payment"),
    ),
    arize=ArizeSpec(
        fail_trace_id="claim_fail",
        healed_trace_id="claim_healed",
        evaluators=(
            "payout_within_policy",
            "root_step_is_intake_not_adjuster",
            "replay_flipped",
        ),
        span_tags=("member_id", "policy_tier", "billed_amount", "payout_amount", "root_step_id"),
        improvement_loop="Eval: adjuster-only fix fails → tune intake extraction prompt → suite pass rate rises",
    ),
)

# ---------------------------------------------------------------------------
# 2. Clinical prior authorization (RESEARCH — Anthropic health + eval suite)
# ---------------------------------------------------------------------------
_CLINICAL_AGENTS = (
    AgentSpec(
        "chart_reader", "CHART_READER", "chart_reader",
        "Extract structured facts from clinical notes (ICD, dates, contraindications)",
        (
            HandoffField("diagnosis", "str", "M54.5", "Primary ICD-10"),
            HandoffField("prior_treatments", "list[date]", "['2025-11-02']", "Prior therapy dates"),
            HandoffField("contraindications", "list[str]", "[]", "Documented contraindications"),
        ),
    ),
    AgentSpec(
        "guideline_check", "GUIDELINE_CHECK", "guideline_check",
        "Map diagnosis + history to payer medical-necessity rules",
        (HandoffField("meets_criteria", "bool", "True", "Payer guideline pass/fail"),),
    ),
    AgentSpec(
        "auth_writer", "AUTH_WRITER", "auth_writer", "Draft prior-auth letter with citations",
        (HandoffField("auth_payload", "json", "{...}", "Structured PA request body"),),
    ),
    AgentSpec(
        "submit", "SUBMIT", "submit", "Submit to payer portal + capture acknowledgment",
        (HandoffField("payer_status", "str", "denied", "approved | denied | pended"),),
    ),
)

CLINICAL = ScenarioDomain(
    id="prior_auth",
    label="clinical · prior authorization",
    tier="research",
    tagline="Wrong prior treatment date poisons guideline approval — symptom at submission, fault at chart extraction.",
    task_template="Prior auth for lumbar MRI · member chart CH-9921 · payer UHC",
    agents=_CLINICAL_AGENTS,
    edges=(
        EdgeSpec("chart_reader", "guideline_check", "{diagnosis, prior_treatments, contraindications}"),
        EdgeSpec("guideline_check", "auth_writer", "{meets_criteria, rule_ids[]}"),
        EdgeSpec("auth_writer", "submit", "{auth_payload}"),
    ),
    parallel_groups=(),
    primary_fault=FaultSpec(
        agent="chart_reader",
        field="prior_treatments",
        bad_value=["2025-01-02"],  # should be 2025-11-02
        good_value=["2025-11-02"],
        symptom="Prior auth denied — insufficient conservative therapy interval (90-day rule)",
    ),
    decoy_fault=FaultSpec(
        agent="auth_writer",
        field="auth_payload",
        bad_value={"wording": "weak"},
        good_value={"wording": "standard"},
        symptom="Letter wording suboptimal — replay at auth_writer does not flip denial",
    ),
    oracle="medical_necessity_justified(chart_facts, payer_rules)",
    langgraph=LangGraphSpec(
        checkpointer="MemorySaver",
        fork_api='update_state(as_node="chart_reader", values={"prior_treatments": ["2025-11-02"]})',
        replay_api="invoke from chart_reader checkpoint",
        state_channels=("chart", "guideline", "auth_draft", "submission"),
    ),
    arize=ArizeSpec(
        fail_trace_id="pa_fail",
        healed_trace_id="pa_healed",
        evaluators=(
            "guideline_grounded_in_chart",
            "earliest_fault_before_symptom",
            "replay_flipped",
        ),
        span_tags=("diagnosis", "prior_treatments", "meets_criteria", "payer_status"),
        improvement_loop="Batch 5–10 fault variants · Arize experiment: chart_reader prompt v1 vs v2",
    ),
    research_notes=(
        "Primary research surface: labeled fault suite + LLM-as-judge evals on clinical grounding. "
        "Optional Deepgram voice intake (speech→JSON fault boundary) for Nintendo track."
    ),
)

# ---------------------------------------------------------------------------
# 3. Procurement / Browserbase (SPONSOR — web tool loop)
# ---------------------------------------------------------------------------
_PROCUREMENT_AGENTS = (
    AgentSpec(
        "spec", "SPEC", "spec", "Normalize purchase request → SKU constraints",
        (
            HandoffField("sku", "str", "H100-80GB", "Required SKU pattern"),
            HandoffField("qty", "int", "500", "Units"),
            HandoffField("max_price", "float", "25000.00", "Budget cap (USD)"),
            HandoffField("vendor_allowlist", "list[str]", "['nvidia.com']", "Approved vendors"),
        ),
    ),
    AgentSpec(
        "browser", "BROWSER", "browser", "Live vendor page scrape via Browserbase/Stagehand",
        (
            HandoffField("page_snapshot", "url", "https://...", "Captured DOM / screenshot ref"),
            HandoffField("sku_selected", "str", "H100-80GB-REF", "SKU read from page"),
            HandoffField("unit_price", "float", "24999.00", "Quoted unit price"),
        ),
    ),
    AgentSpec(
        "compare", "COMPARE", "compare", "Rank quotes against spec",
        (HandoffField("best_offer", "json", "{sku, price, vendor}", "Chosen line item"),),
    ),
    AgentSpec(
        "approver", "APPROVER", "approver", "Policy sign-off under delegation limit",
        (HandoffField("approved", "bool", "True", ""),),
    ),
    AgentSpec(
        "po", "PO", "po", "Issue purchase order",
        (HandoffField("po_id", "str", "PO-4412", "Oracle: SKU must match spec.sku"),),
    ),
)

PROCUREMENT = ScenarioDomain(
    id="procurement_gpu",
    label="procurement · GPU purchase",
    tier="sponsor",
    tagline="Browser read wrong variant — cheapest quote is the wrong SKU.",
    task_template="Purchase 500× H100-80GB under $25k/unit · vendor allowlist",
    agents=_PROCUREMENT_AGENTS,
    edges=(
        EdgeSpec("spec", "browser", "{sku, qty, max_price, vendor_allowlist}"),
        EdgeSpec("browser", "compare", "{sku_selected, unit_price, page_snapshot}"),
        EdgeSpec("compare", "approver", "{best_offer}"),
        EdgeSpec("approver", "po", "{approved, best_offer}"),
    ),
    parallel_groups=(),
    primary_fault=FaultSpec(
        agent="browser",
        field="sku_selected",
        bad_value="H100-80GB-REFURB",
        good_value="H100-80GB",
        symptom="Oracle FAIL: selected SKU mismatch vs spec (refurb ≠ new)",
    ),
    decoy_fault=FaultSpec(
        agent="compare",
        field="best_offer",
        bad_value={"vendor": "gray-market"},
        good_value={"vendor": "nvidia.com"},
        symptom="Compare picked gray-market — fixing compare alone does not flip SKU oracle",
    ),
    oracle="selected_sku_matches_spec(spec.sku, po.line_item.sku)",
    langgraph=LangGraphSpec(
        checkpointer="MemorySaver",
        fork_api='update_state(as_node="browser", values={"sku_selected": "H100-80GB"})',
        replay_api="re-invoke from browser checkpoint with injected SKU",
        state_channels=("spec", "browser_session", "quotes", "approval", "po"),
    ),
    arize=ArizeSpec(
        fail_trace_id="procurement_fail",
        healed_trace_id="procurement_healed",
        evaluators=("selected_sku_matches_spec", "replay_flipped"),
        span_tags=("sku_expected", "sku_selected", "page_url", "unit_price"),
        improvement_loop="LLM-as-judge on span attrs: sku_selected vs sku_expected",
    ),
)

# ---------------------------------------------------------------------------
# 4. SOC incident response (OPS — Sentry + Redis + parallel intel)
# ---------------------------------------------------------------------------
_SOC_AGENTS = (
    AgentSpec(
        "alert_triage", "ALERT_TRIAGE", "alert_triage", "Classify incoming SIEM alert",
        (
            HandoffField("severity", "enum", "medium", "low | medium | high | critical"),
            HandoffField("ioc", "str", "abc123…", "Primary indicator of compromise"),
            HandoffField("affected_hosts", "list[str]", "['srv-12']", "Initial host set"),
        ),
    ),
    AgentSpec(
        "threat_intel", "THREAT_INTEL", "threat_intel", "Parallel enrichment from threat feeds",
        (HandoffField("ioc_verdict", "str", "malicious", "benign | suspicious | malicious"),),
    ),
    AgentSpec(
        "enrich", "ENRICH", "enrich", "Merge triage + intel → target set",
        (HandoffField("confirmed_hosts", "list[str]", "['srv-99']", "Hosts to isolate"),),
    ),
    AgentSpec(
        "contain", "CONTAIN", "contain", "Execute containment playbooks",
        (HandoffField("isolated", "list[str]", "['srv-99']", "Hosts successfully isolated"),),
    ),
    AgentSpec(
        "notify", "NOTIFY", "notify", "Page on-call + ticket",
        (HandoffField("threat_active", "bool", "True", "Oracle: true IOC still beaconing"),),
    ),
)

SOC = ScenarioDomain(
    id="soc_incident",
    label="security · SOC incident",
    tier="ops",
    tagline="Wrong IOC hash → containment targets decoy host — threat still active.",
    task_template="Contain SIEM alert #8812 · suspected C2 beacon",
    agents=_SOC_AGENTS,
    edges=(
        EdgeSpec("alert_triage", "enrich", "{severity, ioc, affected_hosts}"),
        EdgeSpec("alert_triage", "threat_intel", "{ioc}"),
        EdgeSpec("threat_intel", "enrich", "{ioc_verdict, feed_refs[]}"),
        EdgeSpec("enrich", "contain", "{confirmed_hosts}"),
        EdgeSpec("contain", "notify", "{isolated, runbook_id}"),
    ),
    parallel_groups=(("threat_intel",),),  # parallel to enrich path from triage
    primary_fault=FaultSpec(
        agent="alert_triage",
        field="ioc",
        bad_value="deadbeef…",
        good_value="cafebabe…",
        symptom="Containment isolated srv-99 but C2 still active on srv-12 (true IOC)",
    ),
    decoy_fault=FaultSpec(
        agent="contain",
        field="isolated",
        bad_value=["srv-99"],
        good_value=["srv-12"],
        symptom="Contain playbook ran — fixing contain alone doesn't clear oracle",
    ),
    oracle="containment_matches_true_ioc(ioc, isolated_hosts, threat_feed)",
    langgraph=LangGraphSpec(
        checkpointer="MemorySaver",
        fork_api='update_state(as_node="alert_triage", values={"ioc": "<true_hash>"})',
        replay_api="re-run enrich → contain → notify from triage fork",
        state_channels=("alert", "intel", "enrichment", "containment", "comms"),
    ),
    arize=ArizeSpec(
        fail_trace_id="soc_fail",
        healed_trace_id="soc_healed",
        evaluators=("containment_matches_true_ioc", "replay_flipped"),
        span_tags=("severity", "ioc", "confirmed_hosts", "threat_active"),
        improvement_loop="On escalate: Sentry issue auto-filed with root_step + blast_radius + replay_rate",
    ),
)

DOMAINS: tuple[ScenarioDomain, ...] = (INSURANCE, CLINICAL, PROCUREMENT, SOC)
BY_ID: dict[str, ScenarioDomain] = {d.id: d for d in DOMAINS}

HERO_ID = "claim_adjudication"
RESEARCH_ID = "prior_auth"
